agent/agent_worker_test.go

```diff
@@ -106,3 +108,67 @@ func TestDisconnectRetry(t *testing.T) {
 	assert.Regexp(t, regexp.MustCompile(`\[warn\] POST http.*/disconnect: 500 \(Attempt 2/4`), l.Messages[2])
 	assert.Equal(t, "[info] Disconnected", l.Messages[3])
 }
+
+func TestAcquireAndRunJobWaiting(t *testing.T) {
+	exptectedSleeps := []time.Duration{}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		switch req.URL.Path {
+		case "/jobs/waitinguuid/acquire":
+			if req.Header.Get("X-Buildkite-Waiting-As-Locked") != "1" {
+				http.Error(rw, "Expected X-Buildkite-Waiting-As-Locked to be set to 1", http.StatusUnprocessableEntity)
+				return
+			}
+
+			backoff_seq, err := strconv.ParseFloat(req.Header.Get("X-Buildkite-Backoff-Sequence"), 64)
+			if err != nil {
+				backoff_seq = 0
+			}
+			delay := math.Pow(2, backoff_seq)
+			sleep, _ := time.ParseDuration(fmt.Sprintf("%fs", delay))
+
+			if backoff_seq > 4 {
+				cancel()
+			}
+
+			exptectedSleeps = append(exptectedSleeps, sleep)
+
+			rw.Header().Set("Retry-After", fmt.Sprintf("%f", delay))
+			rw.WriteHeader(http.StatusLocked)
+			fmt.Fprintf(rw, `{"message": "Job waitinguuid is not yet eligible to be assinged" }`)
+		default:
+			http.Error(rw, "Not found", http.StatusNotFound)
+		}
+	}))
+	defer server.Close()
+
+	client := api.NewClient(logger.Discard, api.Config{
+		Endpoint: server.URL,
+		Token:    "llamas",
+	})
+
+	l := logger.NewBuffer()
+
+	retrySleeps := make([]time.Duration, 0)
```

Comment:
this is equivalent to 
```Go
retrySleeps := []time.Duration{}
```
and looks a bit odd, but isn't actually a problem, making it a [nit] i guess

Reply:
Done